### PR TITLE
refactor: use tokio::watch for allocations

### LINKED ISSUE #364: Drop eventuals in favor of Tokio watch + timers for `allocations`
Subissue of #333

### PR DESCRIPTION
**Pull request by taslimmuhammed**
Fixes #364 
Hi, please have a look when you're free, and as usual tell me if any changes has to be made

### CODE REVIEW

File: common/src/allocations/monitor.rs [hunk 1]
```diff
@@ -61,30 +63,39 @@ pub fn indexer_allocations(
...
-            warn!(
-                "Failed to fetch active or recently closed allocations for indexer {:?}: {}",
-                indexer_address, err
-            );
-
-            // Sleep for a bit before we retry
-            sleep(interval.div_f32(2.0))
-        },
-    )
+            .map_err(|e| e.to_string());
```
Comment (gusinacio) [inline common/src/allocations/monitor.rs:80]: You dont need to map to strings here.

File: common/src/attestations/signers.rs [hunk 2]
```diff
@@ -129,27 +116,27 @@ mod tests {
 
     #[tokio::test]
     async fn test_attestation_signers_update_with_allocations() {
-        let (mut allocations_writer, allocations) = Eventual::<HashMap<Address, Allocation>>::new();
+        let (allocations_tx, allcoations_rx) = watch::channel(HashMap::new());
```
Comment (gusinacio) [inline common/src/attestations/signers.rs:119]: typo `allocations_rx`

**Review by gusinacio — CHANGES_REQUESTED** [review]
You are getting really good on this. Just some really small things!

If typos and some unnecessary map_errs is the best I could find, you are crushing it.

**Comment by taslimmuhammed** [comment]
thanks for the review[✌️], I'll send updates asap.

**Comment by taslimmuhammed** [comment]
please have a look, BTW do you know review the code just by looking at changes on github or is there any other way🤓

**Review by gusinacio — APPROVED** [review]
LGTM, you are doing incredible!

**Comment by taslimmuhammed** [comment]
Thanks again sir🫡

**Comment by taslimmuhammed** [comment]
BTW I don't know if it's right time to ask, but is there any way i could be part of the team? as an intern or junior or any would suffice, I really love working on this project

**Comment by gusinacio** [comment]
> BTW I don't know if it's right time to ask, but is there any way i could be part of the team? as an intern or junior or any would be great, I really love working on this project

Hey, you could check [here](https://thegraph.com/jobs/) for jobs within Graph Protocol.

I work with Semiotic Labs and while we don't have any position open right now, I suggest you sending your resume [here](https://semiotic.ai/jobs/catch-all).

**Pull request merged**